nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java

```diff
@@ -2987,9 +2990,6 @@ public SDVariable var(INDArray arr) {
     public SDVariable var(String name, @NonNull INDArray arr) {
         if (variables.containsKey(name) && variables.get(name).getVariable().getArr() != null)
             throw new IllegalArgumentException("Another variable with the name " + name + " already exists.");
-        Preconditions.checkState(arr.dataType().isFPType(), "Cannot create variable with non-floating point type:" +
```

Comment:
In what context do non-fp variables make even sense?

Reply:
In my import use case, booleans and strings.

Reply:
eg: We've been supporting strings and booleans for a while and can run string manipulation among other ops that are not numeric. Var shouldn't always have to be just trainable parameters it could be ETL.